extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,376 @@
+import * as azdata from 'azdata';
+import { ImportDataModel } from '../wizard/api/models';
+import * as EventEmitter from 'events';
+import { FlatFileProvider } from '../services/contracts';
+import * as constants from '../common/constants';
+
+const headerLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+	'border-bottom': '1px solid'
+};
+
+const styleLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+};
+
+export class DerivedColumnDialog {
+	private _dialogObject: azdata.window.Dialog;
+	private _doneEmitter: EventEmitter = new EventEmitter();
+	private currentTransformation: string[] = [];
+	private currentDerivedColumnName: string = '';
+	private _view!: azdata.ModelView;
+	private _specifyTransformations: azdata.InputBoxComponent[] = [];
+	private _headerInstructionText: azdata.TextComponent;
+	private _bodyInstructionText: azdata.TextComponent;
+
+
+	private _applyButton!: azdata.window.Button;
+	private _transformationTable!: azdata.DeclarativeTableComponent;
+	private _transformationContainer!: azdata.FlexContainer;
+
+	constructor(private _model: ImportDataModel, private _provider: FlatFileProvider) {
+	}
+
+	public openDialog(): Promise<boolean> {
+		this._applyButton = azdata.window.createButton(constants.previewTransformation);
+		this._applyButton.enabled = false;
+		this._dialogObject = azdata.window.createModelViewDialog(
+			constants.createDerivedColumn,
+			'DerivedColumnDialog',
+			'wide'
+		);
+		this._dialogObject.customButtons = [this._applyButton];
+		this._applyButton.hidden = false;
+		this._dialogObject.okButton.enabled = false;
+
+		let tab = azdata.window.createTab('');
+		tab.registerContent(async (view: azdata.ModelView) => {
+			const columnTableData: azdata.DeclarativeTableCellValue[][] = [];
+			this._model.originalProseColumns.forEach(c => {
+				const tableRow: azdata.DeclarativeTableCellValue[] = [];
+				tableRow.push(
+					{ value: false },
+					{ value: c.columnName }
+				);
+				columnTableData.push(tableRow);
+			});
+			this._view = view;
+			const columnTable = view.modelBuilder.declarativeTable().withProps({
+				columns: [
+					{
+						displayName: '',
+						valueType: azdata.DeclarativeDataType.boolean,
+						isReadOnly: false,
+						showCheckAll: true,
+						width: '20px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft
+					},
+					{
+						displayName: constants.columnTableTitle,
+						valueType: azdata.DeclarativeDataType.string,
+						isReadOnly: true,
+						width: '140px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft
+					}
+				],
+				dataValues: columnTableData,
+				CSSStyles: {
+					'table-layout': 'fixed'
+				}
+			}).component();
+
+			console.log(columnTableData);
```

Comment:
Debugging logging? Generally should avoiding printing to console without any context like this